rest-server/src/models/job.js

```diff
@@ -47,36 +47,22 @@ class Job {
         .end((res) => {
           const resJson = typeof res.body === 'object' ?
               res.body : JSON.parse(res.body);
-          const jobList = [];
-          async.each(resJson.frameworkNames, (frameworkName, callback) => {
-            this.getJobStatus(frameworkName, (jobStatus) => {
-              const jobOverview = {
-                name: frameworkName,
-                username: jobStatus.username,
-                state: jobStatus.state,
-                subState: jobStatus.subState,
-                retries: jobStatus.retries,
-                createdTime: jobStatus.createdTime,
-                completedTime: jobStatus.completedTime,
-                appTrackingUrl: jobStatus.appTrackingUrl,
-                appExitType: jobStatus.appExitType
-              };
-              jobList.push(jobOverview);
-              callback();
-            });
-          }, (err) => {
-            jobList.sort((a, b) => b.createdTime - a.createdTime);
-            next(jobList, err);
+          const jobList = res.summarizedFrameworkInfos.map((frameworkInfo) => {
+            return {
+              name: frameworkInfo.frameworkName,
+              username: frameworkInfo.userName,
+              state: frameworkInfo.frameworkState,
+              retries: frameworkInfo.frameworkRetryPolicyState.retriedCount,
```

Comment:
>retriedCount [](start = 63, length = 12)

Seems we need to use this?

transientNormalRetriedCount +
transientConflictRetriedCount +
nonTransientRetriedCount +
unKnownRetriedCount

Reply:
updated.
what's the relationship between `retriedCount` and those four retriedCounts? I thought you added a sum count

Reply:
I have not added the counter yet. retriedCount is the number against the retrymaxnumber user set. It may be or may not be all the retries depends on whether fancy RetryPolicy is enabled

---
In reply to: [167132637](https://github.com/Microsoft/pai/pull/314#discussion_r167132637) [](ancestors = 167132637)